packages/@storefront/flux-capacitor/test/unit/core/store/middleware.ts

```diff
@@ -427,6 +432,30 @@ suite('Middleware', ({ expect, spy, stub }) => {
     });
   });
 
+  describe('redirectAnalyzer()', () => {
+    it('should invoke the next middleware by default', () => {
+      const action = { type: 'FOO' };
+
+      redirectAnalyzer()(next)(action);
+
+      expect(next).to.be.calledWithExactly(action);
+    });
+
+    // tslint:disable-next-line max-line-length
+    it('should suppress subsequent middleware upon receipt of the START_REDIRECT action, until the DONE_REDIRECT action is received', () => {
+      const startRedirectAction = { type: Actions.START_REDIRECT };
+      const doneRedirectAction = { type: Actions.DONE_REDIRECT };
+
+      redirectAnalyzer()(next)(startRedirectAction);
+      redirectAnalyzer()(next)({ type: 'FOO' });
+      redirectAnalyzer()(next)({ type: 'BAR' });
+      redirectAnalyzer()(next)({ type: 'BAZ' });
+      redirectAnalyzer()(next)(doneRedirectAction);
+
+      expect(next).to.have.callCount(1);
```

Comment:
Also assert which arguments were passed in.

Reply:
This update is complete, resolving.